lib/vagrant-libvirt/driver.rb

```diff
@@ -65,6 +65,21 @@ def system_connection
         @system_connection
       end
 
+      def get_client_domain(mid)
+        begin
+          domain = connection.client.lookup_domain_by_uuid(mid)
+        rescue Libvirt::RetrieveError => e
+          if e.libvirt_code == ProviderLibvirt::Util::ErrorCodes::VIR_ERR_NO_DOMAIN
+            @logger.debug("machine #{mid} not found #{e}.")
+            return nil
+          else
+            raise e
+          end
+        end
+
+        domain
+      end
```

Comment:
Turns out this is returning a different object to `get_domain` below.

Suggest renaming slightly to make it a little more obvious for future me that it's not the same and is retrieving the libvirt object itself. Also could you make this method private for now and can decide in the future to make it publicly accessible.
```suggestion
      def get_libvirt_domain(machine)
        begin
          libvirt_domain = connection.client.lookup_domain_by_uuid(machine.id)
        rescue Libvirt::RetrieveError => e
          if e.libvirt_code == ProviderLibvirt::Util::ErrorCodes::VIR_ERR_NO_DOMAIN
            @logger.debug("machine #{machine.name} not found #{e}.")
            return nil
          else
            raise e
          end
        end

        libvirt_domain
      end
```